document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/client/SmartDocumentsClient.kt

```diff
@@ -201,9 +201,17 @@ class SmartDocumentsClient(
         val documentDataIn = PipedInputStream(documentDataOut)
         val documentDataOutWriter = documentDataOut.writer()
         Executors.newSingleThreadExecutor().execute {
-            write(documentDataOutWriter, inputStream, parsedResponse.documentDataStart, parsedResponse.documentDataEnd)
-            documentDataOutWriter.close()
-            inputStream.close()
+            try {
+                write(
+                    documentDataOutWriter,
+                    inputStream,
+                    parsedResponse.documentDataStart,
+                    parsedResponse.documentDataEnd
+                )
+            } finally {
+                documentDataOutWriter.close()
+                inputStream.close()
```

Comment:
The method that created the stream is responsible for closing it. inputstream is provided, so should not be closed here.
And it Kotlin, you can use `.use`. See https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html